pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1002,8 +1003,14 @@ func (lbaas *LbaasV2) ensureSecurityGroup(clusterName string, apiService *v1.Ser
 	lbSecGroupID, err := groups.IDFromName(lbaas.network, lbSecGroupName)
 	if err != nil {
 		// check whether security group does not exist
-		_, ok := err.(*gophercloud.ErrResourceNotFound)
-		if ok {
+		// The security group does not exist while 'err' is object of gophercloud.ErrResourceNotFound
```

Comment:
maybe we can use a small method? since we are using the same thing 3 times.

Reply:
Yeah, good point.